components/brave_extension/extension/brave_extension/content_cosmetic.ts

```diff
@@ -0,0 +1,335 @@
+// Notify the background script as soon as the content script has loaded.
+// chrome.tabs.insertCSS may sometimes fail to inject CSS in a newly navigated
+// page when using the chrome.webNavigation API.
+// See: https://bugs.chromium.org/p/chromium/issues/detail?id=331654#c15
+// The RenderView should always be ready when the content script begins, so
+// this message is used to trigger CSS insertion instead.
+chrome.runtime.sendMessage({
+  type: 'contentScriptsLoaded'
+})
+
+const parseDomain = require('parse-domain')
+
+const queriedIds = new Set()
+const queriedClasses = new Set()
+
+let notYetQueriedClasses: string[] = []
+let notYetQueriedIds: string[] = []
+
+let backgroundReady: boolean = false
+
+const fetchNewClassIdRules = function () {
+  // Only let the backend know that we've found new classes and id attributes
+  // if  the back end has told us its ready to go and we have at least one new
+  // class or id to query for.
+  if (backgroundReady) {
+    if (notYetQueriedClasses.length !== 0 || notYetQueriedIds.length !== 0) {
+      chrome.runtime.sendMessage({
+        type: 'hiddenClassIdSelectors',
+        classes: notYetQueriedClasses,
+        ids: notYetQueriedIds
+      })
+      notYetQueriedClasses = []
+      notYetQueriedIds = []
+    }
+  } else {
+    setTimeout(fetchNewClassIdRules, 100)
+  }
+}
+
+const handleMutations = function (mutations: any[]) {
+  for (const aMutation of mutations) {
+    if (aMutation.type === 'attribute') {
```

Comment:
I don't think this will work - shouldn't it be checking for `attributes` not `attribute`? We need to use proper TS types and not pass in `any` as the params, which would have caught this. I'm happy to fix this now.

Reply:
@petemill yes from checking https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver that looks right to me too (i.e. changing to `attributes` is correct).